sixtyfps_runtime/corelib/api.rs

```diff
@@ -0,0 +1,272 @@
+        /// ```rust
+        /// sixtyfps::sixtyfps! { MyApp := Window { property <int> foo; /* ... */ } }
+        /// let handle = MyApp::new();
+        /// let handle_weak = handle.as_weak();
+        /// let thread = std::thread::spawn(move || {
+        ///     // ... Do some computation in the thread
+        ///     let foo = 42;
+        ///     # assert!(handle_weak.upgrade().is_none()); // note that upgrade fails in a thread
+        ///     // now forward the data to the main thread using upgrade_in_event_loop
+        ///     handle_weak.upgrade_in_event_loop(move |handle| handle.set_foo(foo));
+        /// });
+        /// # thread.join().unwrap(); return; // don't run the event loop in examples
+        /// handle.run();
+        /// ```
+        #[cfg(feature = "std")]
+        pub fn upgrade_in_event_loop(self, func: impl FnOnce(T) + Send + 'static)
+        where
+            T: 'static,
+        {
+            super::invoke_from_event_loop(move || {
+                if let Some(h) = self.upgrade() {
+                    func(h);
+                }
+            })
+        }
+    }
+
+    // Safety: we make sure in upgrade that the thread is the proper one,
+    // and the VWeak only use atomic pointer so it is safe to clone and drop in another thread
+    #[allow(unsafe_code)]
+    #[cfg(feature = "std")]
+    unsafe impl<T: ComponentHandle> Send for Weak<T> {}
+}
+
+pub use weak_handle::*;
+
+/// Adds the specified function to an internal queue, notifies the event loop to wake up.
+/// Once woken up, any queued up functors will be invoked.
+///
+/// This function is thread-safe and can be called from any thread, including the one
+/// running the event loop. The provided functors will only be invoked from the thread
+/// that started the event loop.
+///
+/// You can use this to set properties or use any other SixtyFPS APIs from other threads,
+/// by collecting the code in a functor and queuing it up for invocation within the event loop.
+///
+/// See also [`Weak::upgrade_in_event_loop`]
+///
+/// # Example
+/// ```rust
+/// sixtyfps::sixtyfps! { MyApp := Window { property <int> foo; /* ... */ } }
+/// let handle = MyApp::new();
+/// let handle_weak = handle.as_weak();
+/// let thread = std::thread::spawn(move || {
+///     // ... Do some computation in the thread
+///     let foo = 42;
+///      // now forward the data to the main thread using invoke_from_event_loop
+///     let handle_copy = handle_weak.clone();
+///     sixtyfps::invoke_from_event_loop(move || handle_copy.unwrap().set_foo(foo));
+/// });
+/// # thread.join().unwrap(); return; // don't run the event loop in examples
+/// handle.run();
+/// ```
+pub fn invoke_from_event_loop(func: impl FnOnce() + Send + 'static) {
+    if let Some(backend) = crate::backend::instance() {
+        backend.post_event(alloc::boxed::Box::new(func))
+    } else {
+        panic!("sixtyfps::invoke_from_event_loop() must be called after starting the event loop.")
```

Comment:
This is new and strictly speaking a behavioural change. When `invoke_from_event_loop` was in the API crate, we could access the default backend, which would be implicitly created. With this function here - and it needs to be in order for `Weak` to call it - we can't do that anymore.

I'm inclined to say that's an acceptable change, but what do you think?

Reply:
```suggestion
        panic!("sixtyfps::invoke_from_event_loop() must be called after the SixtyFPS backend is initialized.")
```

This is unfortunate, but i think it's probably fine because creating any component will also initialize the backend. and this function is most usefull when component is already created.